packages/monitor-v2/src/monitor-polymarket/common.ts

```diff
@@ -0,0 +1,415 @@
+  volumeNum: number;
+  clobTokenIds: [string, string];
+}
+
+interface PolymarketMarketWithAncillaryData extends PolymarketMarket {
+  ancillaryData: string;
+}
+
+export interface PolymarketWithEventData extends PolymarketMarketWithAncillaryData {
+  txHash: string;
+  requester: string;
+  proposer: string;
+  timestamp: number;
+  expirationTimestamp: number;
+  eventTimestamp: number;
+  eventBlockNumber: number;
+  eventIndex: number;
+  proposalTimestamp: number;
+  identifier: string;
+  ancillaryData: string;
+  proposedPrice: string;
+}
+
+interface PolymarketOrderBook {
+  market: string;
+  asset_id: string;
+  bids: { price: string; size: string }[];
+  asks: { price: string; size: string }[];
+  hash: string;
+}
+
+export type Order = { price: number; size: number }[];
+export interface MarketOrderbooks {
+  orderBooks: { bids: Order; asks: Order }[];
+}
+
+export interface StoredNotifiedProposal {
+  txHash: string;
+  question: string;
+  proposedPrice: string;
+  notificationTimestamp: number;
+}
+
+export const formatPriceEvents = async (
+  events: ProposePriceEvent[]
+): Promise<
+  {
+    txHash: string;
+    requester: string;
+    proposer: string;
+    timestamp: number;
+    eventTimestamp: number;
+    eventBlockNumber: number;
+    expirationTimestamp: number;
+    proposalTimestamp: number;
+    identifier: string;
+    ancillaryData: string;
+    proposedPrice: string;
+    eventIndex: number;
+  }[]
+> => {
+  const ooDefaultLiveness = 7200;
+  return Promise.all(
+    events.map(async (event: ProposePriceEvent) => {
+      const block = await event.getBlock();
+      return {
+        txHash: event.transactionHash,
+        requester: event.args.requester,
+        proposer: event.args.proposer,
+        timestamp: event.args.timestamp.toNumber(),
+        eventTimestamp: block.timestamp,
+        eventBlockNumber: event.blockNumber,
+        expirationTimestamp: event.args.expirationTimestamp.toNumber(),
+        proposalTimestamp: event.args.expirationTimestamp.toNumber() - ooDefaultLiveness,
+        identifier: event.args.identifier,
+        ancillaryData: event.args.ancillaryData,
+        proposedPrice: ethers.utils.formatEther(event.args.proposedPrice),
+        eventIndex: event.logIndex,
+      };
+    })
+  );
+};
+
+export const getPolymarketMarkets = async (params: MonitoringParams): Promise<PolymarketMarket[]> => {
+  const whereClause = "question_ID IS NOT NULL" + " AND clob_Token_Ids IS NOT NULL";
+
+  const query = `
+    {
+      markets(where: "${whereClause}") {
+        resolvedBy
+        questionID
+        createdAt
+        question
+        outcomes
+        outcomePrices
+        liquidityNum
+        volumeNum
+        clobTokenIds
+        endDate
+      }
+    }
+  `;
+
+  const { markets: polymarketContracts } = (await request(params.graphqlEndpoint, query)) as any;
+
+  // Remove markets with 1 week old endDate or more. So we only monitor markets that ended in the last week
```

Comment:
How is endDate generated in a market? Is it only set on markets that have resolved? Do all markets have it?

Reply:
I think every market has `endDate` on creation. But let's remove this filter to be 100% we are not missing anything.